aziotctl/aziotctl-common/src/set_log_level.rs

```diff
@@ -0,0 +1,68 @@
+use std::fmt;
+use std::fs::File;
+use std::io;
+use std::io::prelude::*;
+use std::process::Command;
+use std::str::FromStr;
+
+use crate::ServiceDefinition;
+
+#[allow(clippy::module_name_repetitions)]
+/// # Errors
+///
+/// Will return `Err` if the systemd folder for the service does not exist or the user does not have
+/// permission to read it.
+pub fn set_log_level(services: &[&ServiceDefinition], level: LogLevel) -> Result<(), io::Error> {
+    for ServiceDefinition {
+        service,
+        sockets: _,
+    } in services
+    {
+        write_log(service, level)?;
+    }
+    Command::new("systemctl").arg("daemon-reload").output()?;
+
+    println!("Set log level to {} for all services. Run the `system restart` command for the changes to take effect.");
+    Ok(())
+}
+
+fn write_log(service: &str, level: LogLevel) -> Result<(), io::Error> {
+    let filename = format!("/etc/systemd/system/{}.d/log-level.conf", service);
+    let contents = format!(
+        "[Service]
+Environment=AZIOT_LOG={}",
+        level
+    );
+
+    let mut file = File::create(filename)?;
+    file.write_all(contents.as_bytes())?;
+
+    Ok(())
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum LogLevel {
+    Debug,
```

Comment:
There's no `Normal = "normal"`. It's `Info = "info"`.

Also, add `Error = "error"`, `Off = "off"`, `Trace = "trace"` and `Warn = "warn"` for completeness.

(Full list is https://docs.rs/log/0.4.14/src/log/lib.rs.html#362 )

Reply:
Actually, it might be better to add `log` as a dependency, and use the `log::Level` type directly.

It already implements `FromStr` and `fmt::Display`, and is guaranteed to stay in-sync with the values used by the daemons.